### PR TITLE
INS-20279 support private link open search

### PR DESCRIPTION
**Pull request by Cai-Chen**

### CODE REVIEW

File: acc_test/data/valid_opensearch_privatelink_cluster_create.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  api_hostname = "%s"
+}
+
+
+resource "instaclustr_cluster" "validOpenSearchPrivateLink" {
+  cluster_name            = "tf-opensearch-test"
+  data_centre             = "US_WEST_2"
+  sla_tier                = "NON_PRODUCTION"
+  cluster_network         = "192.168.0.0/18"
+  private_network_cluster = false
```
Comment (iclonghoang) [inline acc_test/data/valid_opensearch_privatelink_cluster_create.tf:13]: privateLink requires private network cluster

File: acc_test/data/valid_opensearch_privatelink_cluster_create.tf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  cluster_provider        = {
+    name = "AWS_VPC",
+  }
+  rack_allocation         = {
+    number_of_racks = 3
+    nodes_per_rack  = 2
+  }
+  bundle {
+    bundle  = "OPENSEARCH"
+    version = "1.3.2"
```
Comment (iclonghoang) [inline acc_test/data/valid_opensearch_privatelink_cluster_create.tf:23]: this version has just been disabled when we release 1.3.4
I don't know if we have specifically enabled 1.3.2 for the tf test account, better check this

Comment (Cai-Chen) [reply acc_test/data/valid_opensearch_privatelink_cluster_create.tf:23]: Yeh, i will update this version

File: acc_test/data/valid_opensearch_privatelink_cluster_create.tf (hunk 1 above)
Comment (Cai-Chen) [reply acc_test/data/valid_opensearch_privatelink_cluster_create.tf]: yeh, good catch

**Review by iclonghoang — DISMISSED** [review]
thanks for updating 1.3.4 version for me

**Review by iclonghoang — DISMISSED** [review]

**Review by arjunrajnetapp — APPROVED** [review]

**Review by iclonghoang — APPROVED** [review]

**Review by arjunrajnetapp — APPROVED** [review]

**Pull request merged**